### PR TITLE
Phantom tree

### PR DESCRIPTION
**Pull request by hohserg1**
Generation from supling
Avoiding on cutting mehanics https://gyazo.com/1163785ad5f17a9829ff4667f048a10d

### CODE REVIEW

File: src/main/java/ru/mousecray/endmagic/blocks/dimensional/BlockEnderGrass.java [hunk 1]
```diff
@@ -17,10 +17,11 @@
 import net.minecraft.world.World;
 import ru.mousecray.endmagic.api.blocks.EndSoilType;
 import ru.mousecray.endmagic.api.blocks.IEndSoil;
+import ru.mousecray.endmagic.blocks.BlockTypeBase;
 import ru.mousecray.endmagic.blocks.VariativeBlock;
 import ru.mousecray.endmagic.init.EMBlocks;
 
-public class BlockEnderGrass<GrassType extends Enum<GrassType> & IStringSerializable> extends VariativeBlock<GrassType> implements IEndSoil {
+public class BlockEnderGrass<GrassType extends Enum<GrassType> & IStringSerializable & BlockTypeBase> extends VariativeBlock<GrassType> implements IEndSoil {
```
Comment (mousecray) [inline src/main/java/ru/mousecray/endmagic/blocks/dimensional/BlockEnderGrass.java]: Bug: incompatible with other EnumTypes

File: src/main/java/ru/mousecray/endmagic/blocks/trees/EMLog.java [hunk 2]
```diff
@@ -67,4 +68,15 @@ public boolean canSustainLeaves(IBlockState state, net.minecraft.world.IBlockAcc
     public boolean isWood(net.minecraft.world.IBlockAccess world, BlockPos pos) {
         return true;
     }
+
+    @Override
+    public boolean isFullCube(IBlockState state) {
```
Comment (mousecray) [inline src/main/java/ru/mousecray/endmagic/blocks/trees/EMLog.java]: Must applies only for phantom log

Comment (mousecray) [reply src/main/java/ru/mousecray/endmagic/blocks/trees/EMLog.java]: may be

File: src/main/java/ru/mousecray/endmagic/util/EnderBlockTypes.java [hunk 3]
```diff
@@ -1,90 +1,136 @@
...
-		}
-	    
-	    @Override
+        @Override
+        public String getName() {
+            return name;
+        }
+    }
+
+    public static enum EnderGroundType implements IStringSerializable {
```
Comment (mousecray) [inline src/main/java/ru/mousecray/endmagic/util/EnderBlockTypes.java]: Add implementation of BlockTypeBase

File: src/main/java/ru/mousecray/endmagic/blocks/dimensional/BlockEnderGrass.java (hunk 1 above)
Comment (mousecray) [reply src/main/java/ru/mousecray/endmagic/blocks/dimensional/BlockEnderGrass.java]: But fix for this must exist in another class. Close conversation.

File: src/main/java/ru/mousecray/endmagic/blocks/trees/EMLog.java (hunk 2 above)
Comment (hohserg1) [reply src/main/java/ru/mousecray/endmagic/blocks/trees/EMLog.java]: extracted!

File: src/main/java/ru/mousecray/endmagic/util/EnderBlockTypes.java (hunk 3 above)
Comment (hohserg1) [reply src/main/java/ru/mousecray/endmagic/util/EnderBlockTypes.java]: fixed!

**Pull request merged**